publish_process.js

```diff
@@ -0,0 +1,315 @@
+#!/usr/bin/env node
+
+/**
+Perform the next step to publish the extension.
+ i. Verifies that it's on the master branch. If not so, terminate. Otherwise, pulls the latest master version.
+ii. If the last message on master indicates Version X.Y.Z and it does not have a tag.
+    it means that steps iii. to vii. were executed correctly.
+    Ask whether it's ready to publish (ENTER or y)
+    - if so, create and publish the tag
+    - Otherwise, cancel.
+
+iii. Fetches all tags, and verify that there is no version tag.
+ iv. Asks to bump the patch version (ENTER or p), the minor version (m) or the major version (M)
+  v. Populates the CHANGELOG.md with the most recent commit messages.
+ vi. If a newest version of Dafny is found, ask whether to refer to it as the latest (ENTER or y), or no (n)
+vii. Follows the steps of CONTRIBUTING.
+     Ask to revise the CHANGELOG.md as needed
+     (n or anything else) : Abruptly stop.
+     (ENTER or y) : Creates a branch named "version-X.Y.Z"
+                    Commits the files to this branch with message "Version X.Y.Z"
+                    Push the branch.
+     Indicate that the PR should be merge without changing the commit message.
+     Indicate that the script just needs to be relaunched once the PR is approved and merged.
+*/
+
+const fs = require("fs");
+const { promisify, getSystemErrorMap } = require('util');
+const exec = require('child_process').exec;
+const execAsync = promisify(exec);
+const readline = require('readline');
+const fetch = (url, init) =>
+  import('node-fetch').then(({ default: fetch }) => fetch(url, init));
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+const question = function(input) {
+  return new Promise((resolve, reject) => {
+    rl.question(input, resolve);
+  });
+}
+const changeLogFile = "CHANGELOG.md";
+const dafnyReleasesURL = 'https://api.github.com/repos/dafny-lang/dafny/releases';
+const constantsFile = "src/constants.ts";
+const packageFile = "package.json";
+const packageLockFile = "package-lock.json";
+const ABORTED = "ABORTED";
+const ACCEPT_HINT = "(ENTER)";
+
+function ok(answer) {
+  return answer.toLowerCase() == "y" || answer == "";
+}
+
+async function ensureMaster(retry = true) {
```

Comment:
I find the "call yourself just to retry the first step" idiom really confusing. Consider just extracting out a `getCurrentBranch()` function instead.